File: android/src/main/java/org/axen/flutter/glide/provider/DrawableImageProvider.java
package org.axen.flutter.glide.provider;

import android.content.Context;
import android.graphics.Bitmap;
import android.graphics.drawable.Drawable;

import androidx.annotation.NonNull;

import com.bumptech.glide.Glide;
import com.bumptech.glide.RequestBuilder;

public final class DrawableImageProvider extends AbstractProvider<Drawable> {

    public DrawableImageProvider(@NonNull Context context) {
        super(context);
    }

    @NonNull
    protected RequestBuilder<Drawable> build() {
        return Glide.with(this.context).asDrawable();
    }

}

File: android/src/main/kotlin/org/axen/flutter/glide/provider/DrawableImageProvider.kt
package org.axen.flutter.glide.provider

import android.content.Context
import android.graphics.drawable.Drawable
import com.bumptech.glide.Glide
import com.bumptech.glide.RequestBuilder
import org.axen.flutter.glide.common.AbstractProvider

class DrawableImageProvider(context: Context): AbstractProvider<Drawable>(context) {

    override fun build(): RequestBuilder<Drawable> {
        return Glide.with(context).asDrawable()
    }
}